### ISSUE TITLE
RuntimeError: a Tensor with 32768 elements cannot be converted to Scalar

### ISSUE THREAD

**Issue by astachowiczhabana** [huggingface/optimum-habana#933]
### System Info

```shell
docker image:
vault.habana.ai/gaudi-docker/1.15.1/ubuntu22.04/habanalabs/pytorch-installer-2.2.0:latest

oh:
https://github.com/huggingface/optimum-habana/tree/bb6be54d112bcaed2d0ff8ab10ea66b8578ba0ba
```

### Information

- [X] The official example scripts
- [ ] My own modified scripts

### Tasks

- [X] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

1. Pull latest HabanaAI docker
```
docker pull vault.habana.ai/gaudi-docker/1.15.1/ubuntu22.04/habanalabs/pytorch-installer-2.2.0:latest
```
2. Clone latest optimum-habana
```
apt update; apt install gh; gh repo clone huggingface/optimum-habana
```
3. Reproduce the issue
```
cd optimum-habana-fork/examples/language-modeling/; pip install -r requirements.txt; pip install -e ../..; PT_HPU_LAZY_MODE=0 python ../gaudi_spawn.py \
    --world_size 4 --use_mpi run_lora_clm.py \
    --model_name_or_path Llama-2-7b-chat-hf \
    --dataset_name tatsu-lab/alpaca \
    --bf16 True \
    --output_dir /tmp/model_lora_llama_ddp \
    --num_train_epochs 2 \
    --per_device_train_batch_size 8 \
    --evaluation_strategy "no" \
    --save_strategy "no" \
    --learning_rate 1.0e-05 \
    --warmup_ratio  0.03 \
    --lr_scheduler_type "linear" \
    --max_grad_norm  0.3 \
    --logging_steps 1 \
    --do_train \
    --use_habana \
    --use_lazy_mode False \
    --throughput_warmup_steps 3 \
    --lora_rank=8 \
    --lora_alpha=32 \
    --lora_dropout=0.1 \
    --lora_target_modules "q_proj" "v_proj" \
    --dataset_concatenation \
    --max_seq_length 512 \
    --ddp_bucket_cap_mb 50 \
    --adam_epsilon 1e-08 \
    --low_cpu_mem_usage True 
```

4. Result
```
    main()
  File "/root/optimum-habana/examples/language-modeling/run_lora_clm.py", line 792, in main
    train_result = trainer.train(resume_from_checkpoint=training_args.resume_from_checkpoint)
  File "/root/optimum-habana/optimum/habana/transformers/trainer.py", line 526, in train
    return inner_training_loop(
  File "/root/optimum-habana/optimum/habana/transformers/trainer.py", line 1010, in _inner_training_loop
    self._maybe_log_save_evaluate(tr_loss, _grad_norm, model, trial, epoch, ignore_keys_for_eval)
  File "/root/optimum-habana/optimum/habana/transformers/trainer.py", line 1187, in _maybe_log_save_evaluate
    _grad_norm.item()
RuntimeError: a Tensor with 32768 elements cannot be converted to Scalar
```

Note:
Looks like _grad_nor tensor is incompatible, it's initially created here:
https://github.com/huggingface/optimum-habana/blob/main/optimum/habana/transformers/trainer.py#L984

### Expected behavior

```
{'loss': 1.6187, 'learning_rate': 4.0000000000000003e-07, 'epoch': 0.0, 'memory_allocated (GB)': 13.08, 'max_memory_allocated (GB)': 65.23, 'total_memory_available (GB)': 94.62}                                                                                      
{'loss': 1.6596, 'learning_rate': 8.000000000000001e-07, 'epoch': 0.0, 'memory_allocated (GB)': 13.08, 'max_memory_allocated (GB)': 65.24, 'total_memory_available (GB)': 94.62}                                                                                       
{'loss': 1.7088, 'learning_rate': 1.2000000000000002e-06, 'epoch': 0.01, 'memory_allocated (GB)': 13.08, 'max_memory_allocated (GB)': 65.25, 'total_memory_available (GB)': 94.62}                                                                                     
{'loss': 1.6389, 'learning_rate': 1.6000000000000001e-06, 'epoch': 0.01, 'memory_allocated (GB)': 13.08, 'max_memory_allocated (GB)': 65.25, 'total_memory_available (GB)': 94.62}                                                                                     
{'loss': 1.698, 'learning_rate': 2.0000000000000003e-06, 'epoch': 0.01, 'memory_allocated (GB)': 13.08, 'max_memory_allocated (GB)': 65.27, 'total_memory_available (GB)': 94.62}                                                        
{'loss': 1.6361, 'learning_rate': 2.4000000000000003e-06, 'epoch': 0.01, 'memory_allocated (GB)': 13.08, 'max_memory_allocated (GB)': 65.27, 'total_memory_available (GB)': 94.62}                                                                                     
{'loss': 1.6356, 'learning_rate': 2.8000000000000003e-06, 'epoch': 0.02, 'memory_allocated (GB)': 13.08, 'max_memory_allocated (GB)': 65.27, 'total_memory_available (GB)': 94.62}                                                                                     
{'loss': 1.6271, 'learning_rate': 3.2000000000000003e-06, 'epoch': 0.02, 'memory_allocated (GB)': 13.08, 'max_memory_allocated (GB)': 65.27, 'total_memory_available (GB)': 94.62}                                                                                     
{'loss': 1.6358, 'learning_rate': 3.6000000000000003e-06, 'epoch': 0.02, 'memory_allocated (GB)': 13.08, 'max_memory_allocated (GB)': 65.27, 'total_memory_available (GB)': 94.62}       
```

**Comment by regisss** [comment]
I can reproduce this issue, I'm going to take a closer look.
Out of curiosity, why running this script in eager mode and not in lazy mode?

**Comment by yeonsily** [comment]
@regisss I see that it's because we try to use _grad_norm.item() for non one-element tensor.

Should we change https://github.com/huggingface/optimum-habana/blob/main/optimum/habana/transformers/trainer.py#L1186  similar to below?

```
if _grad_norm is not None and self.accelerator.distributed_type != GaudiDistributedType.FSDP:
          grad_norm = _grad_norm.item() if _grad_norm.size()==1 else _grad_norm.tolist()
else: 
          grad_norm = None
```

**Comment by regisss** [comment]
@astachowiczhabana This should be fixed by #938.
